# Importing Libraries and Dataset

In [3]:
import pandas as pd
import numpy as np
import os

In [4]:
path = r'C:\Users\bhava\Documents\18.02.26_Instacart Basket Analysis'

# 1. Importing new pickle data

In [5]:
ords_prods_merge_new = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ords_prods_merge_clean.pkl'))

In [6]:
ords_prods_merge_new.head()

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,product_name,aisle_id,department_id,prices,_merge,price_range_loc,busiest_day,busiest_days,busiest_period_of_day
0,2539329,1,1,2,8,NaN,196,1,0,Soda,77,7,9.0,both,Mid-range product,Regularly busy,Regularly busy,Average orders
1,2539329,1,1,2,8,NaN,14084,2,0,Organic Unsweetened Vanilla Almond Milk,91,16,12.5,both,Mid-range product,Regularly busy,Regularly busy,Average orders
2,2539329,1,1,2,8,NaN,12427,3,0,Original Beef Jerky,23,19,4.4,both,Low-range product,Regularly busy,Regularly busy,Average orders
3,2539329,1,1,2,8,NaN,26088,4,0,Aged White Cheddar Popcorn,23,19,4.7,both,Low-range product,Regularly busy,Regularly busy,Average orders
4,2539329,1,1,2,8,NaN,26405,5,0,XL Pick-A-Size Paper Towel Rolls,54,17,1.0,both,Low-range product,Regularly busy,Regularly busy,Average orders


In [7]:
ords_prods_merge_new.tail()

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,product_name,aisle_id,department_id,prices,_merge,price_range_loc,busiest_day,busiest_days,busiest_period_of_day
32404854,2977660,206209,13,1,12,7.0,14197,5,1,Tomato Paste,9,9,5.6,both,Mid-range product,Regularly busy,Busiest days,Most orders
32404855,2977660,206209,13,1,12,7.0,38730,6,0,Brownie Crunch High Protein Bar,3,19,5.9,both,Mid-range product,Regularly busy,Busiest days,Most orders
32404856,2977660,206209,13,1,12,7.0,31477,7,0,High Protein Bar Chunky Peanut Butter,3,19,4.2,both,Low-range product,Regularly busy,Busiest days,Most orders
32404857,2977660,206209,13,1,12,7.0,6567,8,0,Chocolate Peanut Butter Protein Bar,3,19,4.9,both,Low-range product,Regularly busy,Busiest days,Most orders
32404858,2977660,206209,13,1,12,7.0,22920,9,0,Roasted & Salted Shelled Pistachios,117,19,1.7,both,Low-range product,Regularly busy,Busiest days,Most orders


In [8]:
ords_prods_merge_new.shape

(32404859, 18)

# 2. Grouping department_id and aggregating order_number by mean

In [9]:
ords_prods_merge_new.groupby('department_id')['order_number'].mean()

department_id
1     15.457838
2     17.277920
3     17.170395
4     17.811403
5     15.215751
6     16.439806
7     17.225802
8     15.340650
9     15.895474
10    20.197148
11    16.170638
12    15.887671
13    16.583536
14    16.773669
15    16.165037
16    17.665606
17    15.694469
18    19.310397
19    17.177343
20    16.473447
21    22.902379
Name: order_number, dtype: float64

# Comparing the subset

In [10]:
df = ords_prods_merge_new[:1000000]

In [11]:
df.groupby('department_id')['order_number'].mean()

department_id
1     14.800024
2     17.091743
3     17.913544
4     17.893092
5     15.214270
6     15.382135
7     17.694027
8     16.458105
9     15.957363
10    20.091818
11    16.482026
12    15.615061
13    16.484023
14    17.524632
15    15.691875
16    18.014071
17    16.150593
18    19.602850
19    17.631340
20    17.138607
21    21.956893
Name: order_number, dtype: float64

## 3. The results from the subset differ slightly from the full dataframe because the subset is not a random sample. It only includes 1000000 of the first portion of the data, which can bias the averages. As a result, the mean order_number values are similar but not identical. This shows that subsets may approximate overall trends but do not perfectly represent the full dataset.

# 4. Loyalty customer

In [12]:
df.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
1                14.800024   1  99
2                17.091743   1  98
3                17.913544   1  99
4                17.893092   1  99
5                15.214270   1  99
6                15.382135   1  99
7                17.694027   1  99
8                16.458105   1  91
9                15.957363   1  99
10               20.091818   1  99
11               16.482026   1  99
12               15.615061   1  99
13               16.484023   1  99
14               17.524632   1  99
15               15.691875   1  99
16               18.014071   1  99
17               16.150593   1  99
18               19.602850   1  99
19               17.631340   1  99
20               17.138607   1  99
21               21.956893   1  97

In [13]:
ords_prods_merge_new.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
1                15.457838   1  99
2                17.277920   1  99
3                17.170395   1  99
4                17.811403   1  99
5                15.215751   1  99
6                16.439806   1  99
7                17.225802   1  99
8                15.340650   1  99
9                15.895474   1  99
10               20.197148   1  99
11               16.170638   1  99
12               15.887671   1  99
13               16.583536   1  99
14               16.773669   1  99
15               16.165037   1  99
16               17.665606   1  99
17               15.694469   1  99
18               19.310397   1  99
19               17.177343   1  99
20               16.473447   1  99
21               22.902379   1  99

# Transforming Data using loc()

In [14]:
# Step 1, 2 and 3 

ords_prods_merge_new['max_order'] = ords_prods_merge_new.groupby(['user_id'])['order_number'].transform(np.max)

C:\Users\bhava\AppData\Local\Temp\ipykernel_8156\1989348669.py:3: FutureWarning: The provided callable <function max at 0x0000027551ABD620> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  ords_prods_merge_new['max_order'] = ords_prods_merge_new.groupby(['user_id'])['order_number'].transform(np.max)


In [15]:
# 5. Loyalty Flags

In [16]:
ords_prods_merge_new.loc[ords_prods_merge_new['max_order'] > 40, 'loyalty_flag'] = 'Loyal customer'

In [17]:
ords_prods_merge_new.loc[(ords_prods_merge_new['max_order'] <= 40) & (ords_prods_merge_new['max_order'] > 10), 'loyalty_flag'] = 'Regular customer'

In [18]:
ords_prods_merge_new.loc[ords_prods_merge_new['max_order'] <= 10, 'loyalty_flag'] = 'New customer'

In [19]:
ords_prods_merge_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32404859 entries, 0 to 32404858
Data columns (total 20 columns):
 #   Column                 Dtype   
---  ------                 -----   
 0   order_id               int64   
 1   user_id                int64   
 2   order_number           int64   
 3   order_day_of_week      int64   
 4   order_hour_of_day      int64   
 5   days_since_last_order  float64 
 6   product_id             int64   
 7   add_to_cart_order      int64   
 8   reordered              int64   
 9   product_name           object  
 10  aisle_id               int64   
 11  department_id          int64   
 12  prices                 float64 
 13  _merge                 category
 14  price_range_loc        object  
 15  busiest_day            object  
 16  busiest_days           object  
 17  busiest_period_of_day  object  
 18  max_order              int64   
 19  loyalty_flag           object  
dtypes: category(1), float64(2), int64(11), object(6)
memory usage: 4.6

In [20]:
ords_prods_merge_new['loyalty_flag'].value_counts(dropna = False)

loyalty_flag
Regular customer    15876776
Loyal customer      10284093
New customer         6243990
Name: count, dtype: int64

In [21]:
ords_prods_merge_new.head()

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,product_name,aisle_id,department_id,prices,_merge,price_range_loc,busiest_day,busiest_days,busiest_period_of_day,max_order,loyalty_flag
0,2539329,1,1,2,8,NaN,196,1,0,Soda,77,7,9.0,both,Mid-range product,Regularly busy,Regularly busy,Average orders,10,New customer
1,2539329,1,1,2,8,NaN,14084,2,0,Organic Unsweetened Vanilla Almond Milk,91,16,12.5,both,Mid-range product,Regularly busy,Regularly busy,Average orders,10,New customer
2,2539329,1,1,2,8,NaN,12427,3,0,Original Beef Jerky,23,19,4.4,both,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer
3,2539329,1,1,2,8,NaN,26088,4,0,Aged White Cheddar Popcorn,23,19,4.7,both,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer
4,2539329,1,1,2,8,NaN,26405,5,0,XL Pick-A-Size Paper Towel Rolls,54,17,1.0,both,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer


In [22]:
ords_prods_merge_new.tail()

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,product_name,aisle_id,department_id,prices,_merge,price_range_loc,busiest_day,busiest_days,busiest_period_of_day,max_order,loyalty_flag
32404854,2977660,206209,13,1,12,7.0,14197,5,1,Tomato Paste,9,9,5.6,both,Mid-range product,Regularly busy,Busiest days,Most orders,13,Regular customer
32404855,2977660,206209,13,1,12,7.0,38730,6,0,Brownie Crunch High Protein Bar,3,19,5.9,both,Mid-range product,Regularly busy,Busiest days,Most orders,13,Regular customer
32404856,2977660,206209,13,1,12,7.0,31477,7,0,High Protein Bar Chunky Peanut Butter,3,19,4.2,both,Low-range product,Regularly busy,Busiest days,Most orders,13,Regular customer
32404857,2977660,206209,13,1,12,7.0,6567,8,0,Chocolate Peanut Butter Protein Bar,3,19,4.9,both,Low-range product,Regularly busy,Busiest days,Most orders,13,Regular customer
32404858,2977660,206209,13,1,12,7.0,22920,9,0,Roasted & Salted Shelled Pistachios,117,19,1.7,both,Low-range product,Regularly busy,Busiest days,Most orders,13,Regular customer


In [23]:
ords_prods_merge_new['loyalty_flag']

0               New customer
1               New customer
2               New customer
3               New customer
4               New customer
                  ...       
32404854    Regular customer
32404855    Regular customer
32404856    Regular customer
32404857    Regular customer
32404858    Regular customer
Name: loyalty_flag, Length: 32404859, dtype: object

In [24]:
ords_prods_merge_new[['user_id', 'loyalty_flag', 'order_number', 'max_order']].head(60)

,user_id,loyalty_flag,order_number,max_order
0,1,New customer,1,10
1,1,New customer,1,10
2,1,New customer,1,10
3,1,New customer,1,10
4,1,New customer,1,10
5,1,New customer,2,10
6,1,New customer,2,10
7,1,New customer,2,10
8,1,New customer,2,10
9,1,New customer,2,10


In [25]:
ords_prods_merge_new.describe()

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,aisle_id,department_id,prices,max_order
count,3.240486e+07,3.240486e+07,3.240486e+07,3.240486e+07,3.240486e+07,3.032876e+07,3.240486e+07,3.240486e+07,3.240486e+07,3.240486e+07,3.240486e+07,3.240486e+07,3.240486e+07
mean,1.710745e+06,1.029372e+05,1.714230e+01,2.738867e+00,1.342515e+01,1.110408e+01,2.559866e+04,8.352547e+00,5.895873e-01,7.119612e+01,9.919792e+00,1.198023e+01,3.305217e+01
std,9.872988e+05,5.946610e+04,1.753532e+01,2.090077e+00,4.246380e+00,8.779064e+00,1.408400e+04,7.127071e+00,4.919087e-01,3.821139e+01,6.281485e+00,4.956554e+02,2.515525e+01
min,2.000000e+00,1.000000e+00,1.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00
25%,8.559470e+05,5.142200e+04,5.000000e+00,1.000000e+00,1.000000e+01,5.000000e+00,1.354400e+04,3.000000e+00,0.000000e+00,3.100000e+01,4.000000e+00,4.200000e+00,1.300000e+01
50%,1.711049e+06,1.026160e+05,1.100000e+01,3.000000e+00,1.300000e+01,8.000000e+00,2.530200e+04,6.000000e+00,1.000000e+00,8.300000e+01,9.000000e+00,7.400000e+00,2.600000e+01
75%,2.565499e+06,1.543890e+05,2.400000e+01,5.000000e+00,1.600000e+01,1.500000e+01,3.794700e+04,1.100000e+01,1.000000e+00,1.070000e+02,1.600000e+01,1.130000e+01,4.700000e+01
max,3.421083e+06,2.062090e+05,9.900000e+01,6.000000e+00,2.300000e+01,3.000000e+01,4.968800e+04,1.450000e+02,1.000000e+00,1.340000e+02,2.100000e+01,9.999900e+04,9.900000e+01


In [26]:
ords_prods_merge_new.groupby('loyalty_flag')['prices'].mean()

loyalty_flag
Loyal customer      10.386336
New customer        13.294670
Regular customer    12.495717
Name: prices, dtype: float64

In [27]:
ords_prods_merge_new.shape

(32404859, 20)

## 5. On average, the prices of products purchased by the different types of customers is quite similar. The mean is regular customers is 12.49 while that of loyal customers is 10.38, making it slightly higher for Regular customers than Loyal Customers.

# 6. Types of Spenders - Spending Flag

In [28]:
ords_prods_merge_new['avg_spend_user'] = ords_prods_merge_new.groupby(['user_id'])['prices'].transform(np.mean)

C:\Users\bhava\AppData\Local\Temp\ipykernel_8156\444204724.py:1: FutureWarning: The provided callable <function mean at 0x0000027551ABE020> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "mean" instead.
  ords_prods_merge_new['avg_spend_user'] = ords_prods_merge_new.groupby(['user_id'])['prices'].transform(np.mean)


In [29]:
ords_prods_merge_new[['user_id', 'prices', 'avg_spend_user']].head(25)

,user_id,prices,avg_spend_user
0,1,9.0,6.367797
1,1,12.5,6.367797
2,1,4.4,6.367797
3,1,4.7,6.367797
4,1,1.0,6.367797
5,1,9.0,6.367797
6,1,3.0,6.367797
7,1,4.4,6.367797
8,1,10.3,6.367797
9,1,4.7,6.367797


### Types

In [30]:
ords_prods_merge_new.loc[ords_prods_merge_new['avg_spend_user'] < 10, 'spending_flag'] = 'Low spender'

In [31]:
ords_prods_merge_new.loc[ords_prods_merge_new['avg_spend_user'] >= 10, 'spending_flag'] = 'High spender'

### Frequency

In [32]:
ords_prods_merge_new['spending_flag'].value_counts(dropna = False)

spending_flag
Low spender     31770614
High spender      634245
Name: count, dtype: int64

In [33]:
ords_prods_merge_new[['user_id', 'avg_spend_user', 'spending_flag']].head(25)

,user_id,avg_spend_user,spending_flag
0,1,6.367797,Low spender
1,1,6.367797,Low spender
2,1,6.367797,Low spender
3,1,6.367797,Low spender
4,1,6.367797,Low spender
5,1,6.367797,Low spender
6,1,6.367797,Low spender
7,1,6.367797,Low spender
8,1,6.367797,Low spender
9,1,6.367797,Low spender


In [34]:
ords_prods_merge_new[['user_id', 'avg_spend_user', 'spending_flag']].tail(25)

,user_id,avg_spend_user,spending_flag
32404834,206209,7.058915,Low spender
32404835,206209,7.058915,Low spender
32404836,206209,7.058915,Low spender
32404837,206209,7.058915,Low spender
32404838,206209,7.058915,Low spender
32404839,206209,7.058915,Low spender
32404840,206209,7.058915,Low spender
32404841,206209,7.058915,Low spender
32404842,206209,7.058915,Low spender
32404843,206209,7.058915,Low spender


### Answer: Given the high number of low spenders i.e. 31770614, it make sense to target these customers for the new marketing campaigns.

# 7. Frequency Flags

In [35]:
ords_prods_merge_new['ord_regularity_median'] = ords_prods_merge_new.groupby(['user_id'])['days_since_last_order'].transform(np.median)

C:\Users\bhava\AppData\Local\Temp\ipykernel_8156\1348421127.py:1: FutureWarning: The provided callable <function median at 0x000002755227BE20> is currently using SeriesGroupBy.median. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "median" instead.
  ords_prods_merge_new['ord_regularity_median'] = ords_prods_merge_new.groupby(['user_id'])['days_since_last_order'].transform(np.median)


In [36]:
ords_prods_merge_new[['user_id', 'prices', 'days_since_last_order', 'ord_regularity_median']].head(25)

,user_id,prices,days_since_last_order,ord_regularity_median
0,1,9.0,NaN,20.5
1,1,12.5,NaN,20.5
2,1,4.4,NaN,20.5
3,1,4.7,NaN,20.5
4,1,1.0,NaN,20.5
5,1,9.0,15.0,20.5
6,1,3.0,15.0,20.5
7,1,4.4,15.0,20.5
8,1,10.3,15.0,20.5
9,1,4.7,15.0,20.5


### Types 

In [37]:
ords_prods_merge_new.loc[ords_prods_merge_new['ord_regularity_median'] > 20, 'frequency_flag'] = 'Non-frequent customer'

In [38]:
ords_prods_merge_new.loc[(ords_prods_merge_new['ord_regularity_median'] > 10) & (ords_prods_merge_new['ord_regularity_median'] <= 20), 'frequency_flag']  = 'Regular customer'

In [39]:
ords_prods_merge_new.loc[ords_prods_merge_new['ord_regularity_median'] <= 10, 'frequency_flag'] = 'Frequent customer'

### Frequency

In [40]:
ords_prods_merge_new['frequency_flag'].value_counts(dropna = False)

frequency_flag
Frequent customer        21559853
Regular customer          7208564
Non-frequent customer     3636437
NaN                             5
Name: count, dtype: int64

In [41]:
ords_prods_merge_new[['user_id', 'days_since_last_order', 'ord_regularity_median', 'frequency_flag']].head(25)

,user_id,days_since_last_order,ord_regularity_median,frequency_flag
0,1,NaN,20.5,Non-frequent customer
1,1,NaN,20.5,Non-frequent customer
2,1,NaN,20.5,Non-frequent customer
3,1,NaN,20.5,Non-frequent customer
4,1,NaN,20.5,Non-frequent customer
5,1,15.0,20.5,Non-frequent customer
6,1,15.0,20.5,Non-frequent customer
7,1,15.0,20.5,Non-frequent customer
8,1,15.0,20.5,Non-frequent customer
9,1,15.0,20.5,Non-frequent customer


In [42]:
ords_prods_merge_new[['user_id', 'days_since_last_order', 'ord_regularity_median', 'frequency_flag']].tail(25)

,user_id,days_since_last_order,ord_regularity_median,frequency_flag
32404834,206209,18.0,22.0,Non-frequent customer
32404835,206209,18.0,22.0,Non-frequent customer
32404836,206209,18.0,22.0,Non-frequent customer
32404837,206209,18.0,22.0,Non-frequent customer
32404838,206209,18.0,22.0,Non-frequent customer
32404839,206209,18.0,22.0,Non-frequent customer
32404840,206209,18.0,22.0,Non-frequent customer
32404841,206209,18.0,22.0,Non-frequent customer
32404842,206209,18.0,22.0,Non-frequent customer
32404843,206209,18.0,22.0,Non-frequent customer


### Answer: There are 21559853 frequent customers who can be targeted on the Instacart App to send them notifications for ads and offers.

# Additional data cleaning from Exercise 4.9 

In [43]:
ords_prods_merge_new.head()

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,product_name,...,price_range_loc,busiest_day,busiest_days,busiest_period_of_day,max_order,loyalty_flag,avg_spend_user,spending_flag,ord_regularity_median,frequency_flag
0,2539329,1,1,2,8,NaN,196,1,0,Soda,...,Mid-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
1,2539329,1,1,2,8,NaN,14084,2,0,Organic Unsweetened Vanilla Almond Milk,...,Mid-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
2,2539329,1,1,2,8,NaN,12427,3,0,Original Beef Jerky,...,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
3,2539329,1,1,2,8,NaN,26088,4,0,Aged White Cheddar Popcorn,...,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
4,2539329,1,1,2,8,NaN,26405,5,0,XL Pick-A-Size Paper Towel Rolls,...,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer


In [46]:
ords_prods_merge_new.loc[ords_prods_merge_new['prices'] > 100]

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,product_name,...,price_range_loc,busiest_day,busiest_days,busiest_period_of_day,max_order,loyalty_flag,avg_spend_user,spending_flag,ord_regularity_median,frequency_flag
1576,912404,17,12,2,14,5.0,21553,5,0,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Regularly busy,Regularly busy,Most orders,40,Regular customer,108.648299,High spender,5.0,Frequent customer
1638,603376,17,22,6,16,4.0,21553,3,1,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Regularly busy,Regularly busy,Most orders,40,Regular customer,108.648299,High spender,5.0,Frequent customer
16522,3264360,135,2,2,21,13.0,21553,6,0,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Regularly busy,Regularly busy,Average orders,4,New customer,1154.792308,High spender,12.0,Regular customer
16528,892534,135,3,0,8,12.0,21553,3,1,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Busiest day,Busiest days,Average orders,4,New customer,1154.792308,High spender,12.0,Regular customer
53672,229704,342,8,1,19,30.0,21553,9,0,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Regularly busy,Busiest days,Average orders,16,Regular customer,114.426619,High spender,23.0,Non-frequent customer
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
32320511,3172853,205650,18,1,9,7.0,21553,17,1,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Regularly busy,Busiest days,Most orders,25,Regular customer,351.141618,High spender,6.0,Frequent customer
32347717,2504315,205818,3,5,15,3.0,21553,13,0,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Regularly busy,Regularly busy,Most orders,25,Regular customer,170.073770,High spender,13.0,Regular customer
32347727,1108388,205818,5,4,5,1.0,21553,5,1,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Least busy,Slowest days,Fewest orders,25,Regular customer,170.073770,High spender,13.0,Regular customer
32380527,1916142,206049,1,2,17,NaN,21553,2,0,Lowfat 2% Milkfat Cottage Cheese,...,High-range product,Regularly busy,Regularly busy,Average orders,5,New customer,938.031250,High spender,7.0,Frequent customer


In [47]:
ords_prods_merge_new.loc[ords_prods_merge_new['prices'] >100, 'prices'] = np.nan

In [48]:
ords_prods_merge_new.head()

,order_id,user_id,order_number,order_day_of_week,order_hour_of_day,days_since_last_order,product_id,add_to_cart_order,reordered,product_name,...,price_range_loc,busiest_day,busiest_days,busiest_period_of_day,max_order,loyalty_flag,avg_spend_user,spending_flag,ord_regularity_median,frequency_flag
0,2539329,1,1,2,8,NaN,196,1,0,Soda,...,Mid-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
1,2539329,1,1,2,8,NaN,14084,2,0,Organic Unsweetened Vanilla Almond Milk,...,Mid-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
2,2539329,1,1,2,8,NaN,12427,3,0,Original Beef Jerky,...,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
3,2539329,1,1,2,8,NaN,26088,4,0,Aged White Cheddar Popcorn,...,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer
4,2539329,1,1,2,8,NaN,26405,5,0,XL Pick-A-Size Paper Towel Rolls,...,Low-range product,Regularly busy,Regularly busy,Average orders,10,New customer,6.367797,Low spender,20.5,Non-frequent customer


In [56]:
ords_prods_merge_new['prices'].max()

25.0

# 9. Export

In [57]:
ords_prods_merge_new.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'ords_prods_merge_new.pkl'))